### PR TITLE
Handled absence of auth keys exception

### LINKED ISSUE #5588: Facebook Login Should throw an error when auth keys unavailable
**Describe the bug**
Whenever a user clicks on Login with Facebook, he should be notified if the keys are not set in the admin settings.

**To Reproduce**
1. Go to 'Login Page'
2. Click on 'Login with Facebook'
3. Doesn't do anything

**Expected behavior**
It should notify the user that the keys are not set, otherwise redirect to facebook oauth login page.

**Additional context**
Working on this

### PR DESCRIPTION
**Pull request by mrsaicharan1**
Fixes #5588 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
- Whenever the keys are not configured on the server, an internal server error is thrown, saying that the keys are not present.

#### Changes proposed in this pull request:

- Added exception handling for no Client id and secret

### CODE REVIEW

File: app/api/auth.py [hunk 1]
```diff
@@ -94,6 +94,10 @@ def login_user(provider, auth_code):
             'client_secret': provider_class.get_client_secret(),
             'code': auth_code
         }
+        if not payload['client_id']:
+            return 'Client_id not configured on server', 500
+        if not payload['client_secret']:
+            return 'Client_secret no configured on server', 500
```
Comment (iamareebjamal) [inline app/api/auth.py]: 500 is internal server error. You should never send 500 to clients, it is an unexpected error

Secondly, just OR the condition and send a proper JSON telling which social auth is not enabled

JSON should be consistent with JSONAPI spec and in line with other errors returned from the server

File: app/api/helpers/csv_jobs_util.py [hunk 2]
```diff
@@ -64,6 +65,10 @@ def export_sessions_csv(sessions):
             column.append(strip_tags(session.short_abstract) if session.short_abstract else '')
             column.append(session.created_at if session.created_at else '')
             column.append('Yes' if session.is_mail_sent else 'No')
+            column.append(session.level)
+            column.append(session.state)
+            column.append(session.type)
+            column.append(len(session.long_abstract))
```
Comment (iamareebjamal) [inline app/api/helpers/csv_jobs_util.py]: Unrelated changes

File: app/api/auth.py (hunk 1 above)
Comment (mrsaicharan1) [reply app/api/auth.py]: @iamareebjamal Yeah got it. Will remove the 500 error and just throw an error stating that the facebook login hasn't been configured.

File: app/api/auth.py [hunk 3]
```diff
@@ -94,6 +94,8 @@ def login_user(provider, auth_code):
             'client_secret': provider_class.get_client_secret(),
             'code': auth_code
         }
+        if not payload['client_id'] or payload['client_secret']:
+            return make_response(jsonify(message="Facebook Login hasn't been configured"))
```
Comment (iamareebjamal) [inline app/api/auth.py]: Please paste what the response will look like. Also I think this will send 200 OK as a code which is definitely worse than 500

Comment (mrsaicharan1) [reply app/api/auth.py]: I think you're right. But what code would you suggest me to return?

Comment (iamareebjamal) [reply app/api/auth.py]: 501 Not Implemented

Comment (iamareebjamal) [reply app/api/auth.py]: Please paste the JSON response

Comment (mrsaicharan1) [reply app/api/auth.py]: 501 done. Will paste the response.

**Comment by iamareebjamal** [comment]
Please paste the JSON response

**Comment by mrsaicharan1** [comment]
@iamareebjamal 
`{
  "message": "Facebook has not been configured yet"
}`

We need to output the code too right?

**Comment by iamareebjamal** [comment]
No, but it should be consistent with other JSONAPI error messages and it is not

**Comment by mrsaicharan1** [comment]
@iamareebjamal Will take care of the JSON APISPEC and update.

**Comment by mrsaicharan1** [comment]
@iamareebjamal The Output now shows 
`{"errors": [{"status": 501, "source": {"source": ""}, "title": "Not Implemented", "detail": "Facebook Login Not Configured"}], "jsonapi": {"version": "1.0"}}`.
Also implemented it in the helpers file.

**Comment by iamareebjamal** [comment]
Fix hound and codacy issue

**Comment by mrsaicharan1** [comment]
@iamareebjamal Fixed all issues.

**Comment by mrsaicharan1** [comment]
@iamareebjamal Had to raise an in-built exception(was returning) instead of redefining NotImplementedError and returning it(to avoid codacy issues). Otherwise, we cannot JSON-serialize inbuilt NotImplementedErrors.

File: app/api/helpers/errors.py [hunk 4]
```diff
@@ -15,7 +15,6 @@ class ErrorResponse:
 
     def __init__(self, source, detail, title=None, status=None):
         """Initialize a jsonapi ErrorResponse Object
-
```
Comment (iamareebjamal) [inline app/api/helpers/errors.py]: Unneeded

File: app/api/helpers/errors.py [hunk 5]
```diff
@@ -14,8 +14,8 @@ class ErrorResponse:
     headers = {'Content-Type': 'application/vnd.api+json'}
 
     def __init__(self, source, detail, title=None, status=None):
-        """Initialize a jsonapi ErrorResponse Object
-
+        """
+        Initialize a jsonapi ErrorResponse Object
```
Comment (iamareebjamal) [inline app/api/helpers/errors.py]: Unneeded change, there should be no changes in this file

Comment (mrsaicharan1) [reply app/api/helpers/errors.py]: @iamareebjamal Will revert it back to the original file

File: app/api/helpers/errors.py [hunk 6]
```diff
@@ -77,4 +77,4 @@ class BadRequestError(ErrorResponse):
     Default class for 400 Error
     """
     status = 400
-    title = 'Bad Request'
+    title = 'Bad Request'
```
Comment (iamareebjamal) [inline app/api/helpers/errors.py]: Still there's a change

**Comment by mrsaicharan1** [comment]
@iamareebjamal I have reverted the file change. Also, could you tell me if that if the core problem has been taken care of in this PR? Raising the error in compliance to the JSON API specs?

**Comment by mrsaicharan1** [comment]
@iamareebjamal The IDE which I was using was making automatic changes to the file when I was trying to restore it. It's back to normal now. Are the above change fine?
